Neos.Flow/Classes/Security/Authorization/Privilege/Entity/Doctrine/PropertyConditionGenerator.php

```diff
@@ -482,7 +482,7 @@ protected function getConstraintStringForSimpleProperty(SQLFilter $sqlFilter, $p
                     }
                 }
                 $parameter = implode(',', $parameters);
-            } elseif ($this->getRawParameterValue($operandDefinition) !== null) {
+            } elseif ($this->getRawParameterValue($operandDefinition) !== null && $this->getRawParameterValue($operandDefinition) !== []) {
```

Comment:
Hm, tbh I don't really understand this check. The `rawParameterValue` is checked, but then the `$operandDefinition` is given to the SqlFilter, which only accepts a string with `getParameter` - so shouldn't the check be `is_string($operandDefinition)`?

Reply:
I'm not quite sure. The code is pretty old and there aren't any tests for this class. I just added something quite specific that fixes my use case that hopefully doesn't break anything else.

Reply:
Yep, that's what I'm unsure about, because that affects more cases than just the `in` operator I can't even follow the existing logic. Maybe @kdambekalns or @kitsunet knows a bit more about this logic?

Reply:
```suggestion
            } elseif ($this->getRawParameterValue($operandDefinition) !== null && ($this->operator !== 'in' || $this->getRawParameterValue($operandDefinition) !== [])) {
```
I think this is how it needs to be to really only affect the `in` operation, where an empty array as operand will not work as expected. Now, this is really hard to read though. Some more code style love needed desperately.

Reply:
```suggestion
            } elseif (!($this->getRawParameterValue($operandDefinition) === null || ($this->operator === 'in' && $this->getRawParameterValue($operandDefinition) === []))) {
```
That's probably better, negating every single condition is hard to grasp.